packages/tools/rpi-eeprom/sources/rpi-eeprom-update

```diff
@@ -0,0 +1,377 @@
+      return
+   fi
+
+   # Bootloader EEPROM chip-select is muxed with audio pin so disable audio
+   # LDO first to avoid sending noise to analog audio.
+   /opt/vc/bin/vcmailbox 0x00030056 4 4 0 > /dev/null || true
+   dtparam audio=off
+
+   # Switch the SPI pins to boot EEPROM
+   dtoverlay spi-gpio40-45
+   modprobe spidev
+   modprobe spi-bcm2835
+
+   prepareImage "${eeprom_image}"
+
+   echo "Applying update ${eeprom_image}"
+   flashrom -p "linux_spi:dev=/dev/spidev0.0,spispeed=${SPI_SPEED}" -w "${TMP_EEPROM_IMAGE}" || die "flashrom EEPROM update failed"
+
+   dtparam -R spi-gpio40-45
+   dtparam audio=on
+   /opt/vc/bin/vcmailbox 0x00030056 4 4 1 > /dev/null || true
+}
+
+# Use the version reported by the loaded EEPROM instead of attempting to retrieve
+# this via flashrom to avoid unnecessary audio glitches.
+CURRENT_VERSION=
+getCurrentVersion() {
+   if vcgencmd bootloader_version | grep -q timestamp; then
+      CURRENT_VERSION=$(vcgencmd bootloader_version | grep timestamp | awk '{print $2}')
+      if [ "${CURRENT_VERSION}" = "0" ]; then
+         # If a timestamp of zero is returned then it's new firmware but an
+         # old bootloader. Assume bootloader v0
+         CURRENT_VERSION="${FIRST_VERSION}"
+      fi
+   else
+      # New bootloader / old firmware ? Try to parse the date
+      CURRENT_VERSION=$(date -u +%s --date "$(vcgencmd bootloader_version | head -n1)")
+   fi
+
+   # Failed to parse the version. Default to the initial production release.
+   if [ -z "${CURRENT_VERSION}" ]; then
+      CURRENT_VERSION="${FIRST_VERSION}"
+   fi
+}
+
+# Set to the latest critical firmware version
+CRITICAL_UPDATE_IMAGE=""
+CRITICAL_UPDATE_VERSION=0
+getLatestCriticalUpdate() {
+   CRITICAL_UPDATE_VERSION=0
+   match=".*/pieeprom-[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9].bin"
+   latest="$(find "${FIRMWARE_CRITICAL_DIR}" -maxdepth 1 -type f -size "${EEPROM_SIZE}c" -regex "${match}" | sort -r | head -n1)"
+   if [ -f "${latest}" ]; then
+      CRITICAL_UPDATE_VERSION=$(strings "${latest}" | grep BUILD_TIMESTAMP | sed 's/.*=//g')
+      CRITICAL_UPDATE_IMAGE="${latest}"
+   fi
+}
+
+checkDependencies() {
+   if [ ! -d "${FIRMWARE_CRITICAL_DIR}" ]; then
+      die "Bootloader critical updates directory ${FIRMWARE_CRITICAL_DIR} not found."
+   fi
+
+   if vcgencmd bootloader_config | grep -qi "Command not registered"; then
+      die "vcgencmd: bootloader_config. not supported. Please update VC firmware"
+   fi
+
+   if ! flashrom --version > /dev/null 2>&1; then
+      [ "${USE_FLASHROM}" = 0 ] || die "flashrom not found."
+   fi
+
+   if [ "${USE_FLASHROM}" = 0 ]; then
+      [ -f "${RECOVERY_BIN}" ] || die "${RECOVERY_BIN} not found"
+   fi
+}
+
+usage() {
+cat <<EOF
+rpi-eeprom-update [options]... [FILE]
+   Checks whether there Raspberry Pi bootloader EEPROM is up to date and
+   optionally updates the EEPROM.
+
+   A backup of the current EEPROM image is written to ${FIRMWARE_BACKUP_DIR}
+   before applying the update.
```

Comment:
@timg236 only the config is being written to the backup directory - the eeprom image is _not_ written to the backup (so the message above is not correct).

Reply:
Thanks, I'll update the text